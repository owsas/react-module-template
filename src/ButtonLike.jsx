import React from 'react';
import PropTypes from 'prop-types';
import ThumbsUpIcon from 'react-icons/lib/fa/thumbs-up';

import ButtonInteract from './ButtonInteract';

const ButtonLike = props => (
  <ButtonInteract
    type="likes"
    text="Me gusta"
    objectId={props.objectId}
    interactionKey={props.interactionKey}
    className={props.className}
    textDone="Te gusta"
    icon={<ThumbsUpIcon />}
    onLogin={props.onLogin}
  />
);

ButtonLike.propTypes = {
  onLogin: PropTypes.func.isRequired,
  objectId: PropTypes.string.isRequired,
  interactionKey: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
};

export default ButtonLike;

