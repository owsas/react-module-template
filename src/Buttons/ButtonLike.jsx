import React from 'react';
import PropTypes from 'prop-types';
import IconDone from 'react-icons/lib/ti/heart-full-outline';
import IconNotDone from 'react-icons/lib/ti/heart-outline';

import ButtonAct from './ButtonAct';

const ButtonLike = props => (
  <ButtonAct
    verb="liked"
    done={() => <span><IconDone /> Te gusta</span>}
    notDone={() => <span><IconNotDone /> Me gusta</span>}
    {...props}
  />
);

ButtonLike.defaultProps = ButtonAct.defaultProps;
ButtonLike.propTypes = {
  actor: PropTypes.string.isRequired,
  object: PropTypes.string,
  onSave: PropTypes.func,
  onDestroy: PropTypes.func,
};

export default ButtonLike;
