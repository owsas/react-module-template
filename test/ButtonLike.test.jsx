import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import ButtonLike from '../src/ButtonLike';

test('should match snapshot', () => {
  const m = shallow(<ButtonLike
    onLogin={() => {}}
    className="Promo"
    objectId="E5SyTySkx5"
    interactionKey="promoId"
  />);

  expect(toJSON(m)).toMatchSnapshot();
});
