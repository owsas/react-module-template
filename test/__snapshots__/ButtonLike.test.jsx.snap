// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<ButtonInteract
  className="Promo"
  icon={<FaThumbsUp />}
  interactionKey="promoId"
  objectId="E5SyTySkx5"
  onLogin={[Function]}
  text="Me gusta"
  textDone="Te gusta"
  type="likes"
/>
`;
